databricks_mcp/src/databricks_mcp/mcp.py

```diff
@@ -32,6 +35,76 @@
 }
 
 
+def _handle_mcp_errors(func: Callable) -> Callable:
+    """Decorator to handle MCP connection errors for sync wrapper methods."""
+
+    def _process_mcp_error(client_instance, error: Exception) -> None:
+        """Process and enhance MCP connection errors with better context."""
+        # For Databricks-managed MCP servers, no special error processing needed
+        if client_instance._get_databricks_managed_mcp_url_type() is not None:
+            raise error
+
+        try:
+            headers = client_instance.client.config.authenticate()
+            authorization_header = headers["Authorization"]
+            token = authorization_header.split("Bearer ")[1]
+
+            headers = {
+                "Content-Type": "application/json",
+                "Accept": "application/json, text/event-stream",
+                "Authorization": f"Bearer {token}",
+            }
+            payload = json.dumps(
+                {
+                    "jsonrpc": "2.0",
+                    "method": "initialize",
+                    "id": "1av",
+                    "params": {
+                        "clientInfo": {"name": "test-client", "version": "1.0"},
+                        "protocolVersion": "2025-06-18",
+                        "capabilities": {},
+                    },
+                }
+            )
+            response = requests.request(
+                "POST",
+                client_instance.server_url,
+                headers=headers,
+                data=payload,
+                allow_redirects=False,
+            )
+        except Exception as e:
+            # Error during processing the error, re-raise the original error
+            raise error from None
+
+        # Auth errors to Databricks Apps are redirected to a login page; a 302 often indicates an auth issue
+        if response.status_code == 302:
+            raise PermissionError(
+                "Access denied to the MCP server. When accessing an MCP server hosted on a Databricks App please ensure you are using a valid OAuth token. "
+                "If using a Service Principal, ensure that the service principal has query permissions on the Databricks App. "
+                "For more information refer to the documentation here: "
+                "https://docs.databricks.com/aws/en/generative-ai/mcp/custom-mcp?language=Local+environment#connect-to-the-custom-mcp-server"
+            ) from None
+        # Not finding a `/initialize` endpoint means the MCP server is not running or the endpoint is not correct
+        elif response.status_code == 404:
```

Comment:
I think we just return back MCPError for these use cases instead of response.status_code being different? That should help show the errors a bit better? What do you think?